new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/Command.java

```diff
@@ -46,12 +46,24 @@ public boolean execute() {
           params.command(), 
           params.redirect(),
           params.env());
-      if (result.exitValue() != 0) {
-        logger.warning("The command execution failed with the result: {0}", result);
+      if (params.saveStdOut()) {
+        params.stdOut(result.stdout());
+      }
+
+      // check both exitValue and stderr since sometimes the exitValue was 0 even when
+      // the test execution failed.
```

Comment:
Dongbo, can you give details of when WIT returned a 0 exit code even though image creation failed?  Our tests should not work around bugs in the product.

Reply:
This is not only applicable to the WIT commands This method is used for all command execution such as running a curl command to download the tool, or creating a zip file. I tested a couple of error conditions such as wrong download url, and somehow the exitValue was zero. So I added the additional checks to make sure the code catches the error condition. I myself found this behavior strange/unexpected. Let me test those error conditions one more time tomorrow and see if I did something wrong last week when I did the testing.

Reply:
Jus to be clear - I have not seen that a WIT command returned zero exitValue when there was a problem in command execution.

Reply:
I did more testing. In the most of the cases, the exitValue does reflect the execution status. Only in the new getRealVersionIfNeeded method, there were cases that the exitValue is still zero when there was a problem in one of the commands that pipelined together. In those cases, the later check in Installer.java line#99 will catch it and log a severe message. I'll change the code back to only check exitValue.